policy/modules/system/systemd.if

```diff
@@ -2256,3 +2256,80 @@ interface(`systemd_delete_private_tmp',`
 	allow $1 systemd_private_tmp_type:lnk_file delete_lnk_file_perms;
 	allow $1 systemd_private_tmp_type:sock_file delete_sock_file_perms;
 ')
+
+#######################################
+## <summary>
+##	Create objects in the pid directory
+##	with a private type with a type transition.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`systemd_userdbd_data_pid_filetrans',`
+	gen_require(`
+		type init_var_run_t;
+		type systemd_userdbd_data_t;
+	')
+
+	filetrans_pattern($1, init_var_run_t, systemd_userdbd_data_t, dir, "userdb")
+')
+
+#######################################
+## <summary>
+##	Manage systemd-userdbd data sockets.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`systemd_userdbd_data_manage_sock_files',`
+	gen_require(`
+		type systemd_userdbd_data_t;
+	')
+
+	manage_sock_files_pattern($1, systemd_userdbd_data_t, systemd_userdbd_data_t);
+')
+
+#######################################
+## <summary>
+##	Allow domain rw access to systemd-userdbd data
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`systemd_userdbd_rw_data',`
+	gen_require(`
+		type systemd_userdbd_data_t;
+	')
+
+	list_dirs_pattern($1, systemd_userdbd_data_t, systemd_userdbd_data_t)
+	read_lnk_files_pattern($1, systemd_userdbd_data_t, systemd_userdbd_data_t)
+	rw_sock_files_pattern($1, systemd_userdbd_data_t, systemd_userdbd_data_t)
+')
```

Comment:
o now i see... this interface is not what the name suggests. Still though, i think this too broad

Reply:
... but just a little. I dont think the sock files have to be readable

Reply:
Yes just for regular unix stream connect i think. (but seems the entities that write to the dynamicuser and homed sock files to stream oonnect to init_t and homed_t respectively also also list the /run/systemd/userdb directory)

Reply:
the read_lnk_files is only for entities that connect to userdbd via the multiplexer socket ( there is a symlink in /run/systemd/userdb  that point to the multiplexer sock ffile. (the multiplexer sock file is only use by pam clients (inclusing authlogin_login_pgm_domain)  to unix stream connect to userdbd

Reply:
So all in all i the read permission is removed from sock_file then this interface could in theory be used like so:
```
interface(`userdb_unix_stream_connect_dynamicuser',`
# io.systemd.DynamicUser
allow $1 init_t:unix_stream_socket connectto;
systemd_userdbd_rw_data($1)
')
interface(`userdb_unix_stream_connect',`
# io.systemd.Multiplexer and io.systemd.NameServiceSwitch
allow $1 userdbd_t:unix_stream_socket connectto;
systemd_userdbd_rw_data($1)
')
interface(`userdb_unix_stream_connect',`
# io.systemd.Home
allow $1 homed_t:unix_stream_socket connectto;
systemd_userdbd_rw_data($1)
')
```
obviously the above example is not proper but its just to give an indication of how it is grouped in my experience.